tests/unit/pidfile/pidfile_test.c

```diff
@@ -0,0 +1,203 @@
+
+    rc = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EPROTO, rc);
+}
+
+MTF_DEFINE_UTEST(pidfile_test, deserialize_missing_alias)
+{
+    int rc;
+    char home[PATH_MAX];
+    struct pidfile content;
+
+    snprintf(home, sizeof(home), "%s/deserialize-missing-alias", config_root);
+
+    rc = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, rc);
+}
+
+MTF_DEFINE_UTEST(pidfile_test, deserialize_missing_pid)
+{
+    int rc;
+    char home[PATH_MAX];
+    struct pidfile content;
+
+    snprintf(home, sizeof(home), "%s/deserialize-missing-pid", config_root);
+
+    rc = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, rc);
+}
+
+MTF_DEFINE_UTEST(pidfile_test, deserialize_missing_socket)
+{
+    int rc;
+    char home[PATH_MAX];
+    struct pidfile content;
+
+    snprintf(home, sizeof(home), "%s/deserialize-missing-socket", config_root);
+
+    rc = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, rc);
+}
+
+MTF_DEFINE_UTEST(pidfile_test, deserialize_missing_socket_path)
+{
+    int rc;
+    char home[PATH_MAX];
+    struct pidfile content;
+
+    snprintf(home, sizeof(home), "%s/deserialize-missing-socket-path", config_root);
+
+    rc = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, rc);
+}
+
+MTF_DEFINE_UTEST(pidfile_test, deserialize_root_wrong_type)
+{
+    int rc;
+    char home[PATH_MAX];
+    struct pidfile content;
+
+    snprintf(home, sizeof(home), "%s/deserialize-root-wrong-type", config_root);
+
+    rc = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, rc);
+}
+
+MTF_DEFINE_UTEST(pidfile_test, deserialize_alias_wrong_type)
+{
+    int rc;
+    char home[PATH_MAX];
+    struct pidfile content;
+
+    snprintf(home, sizeof(home), "%s/deserialize-alias-wrong-type", config_root);
+
+    rc = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, rc);
+}
+
+MTF_DEFINE_UTEST(pidfile_test, deserialize_pid_wrong_type)
+{
+    int rc;
+    char home[PATH_MAX];
+    struct pidfile content;
+
+    snprintf(home, sizeof(home), "%s/deserialize-pid-wrong-type", config_root);
+
+    rc = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, rc);
+}
+
+MTF_DEFINE_UTEST(pidfile_test, deserialize_socket_wrong_type)
+{
+    int rc;
+    char home[PATH_MAX];
+    struct pidfile content;
+
+    snprintf(home, sizeof(home), "%s/deserialize-socket-wrong-type", config_root);
+
+    rc = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, rc);
+}
+
+MTF_DEFINE_UTEST(pidfile_test, deserialize_socket_path_wrong_type)
+{
+    int rc;
+    char home[PATH_MAX];
+    struct pidfile content;
+
+    snprintf(home, sizeof(home), "%s/deserialize-socket-path-wrong-type", config_root);
+
+    rc = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, rc);
+}
+
+MTF_DEFINE_UTEST(pidfile_test, deserialize_home_dne)
+{
+    int rc;
+    struct pidfile content;
+
+    rc = pidfile_deserialize("/this-does-not-exist", &content);
+    ASSERT_EQ(ENOENT, rc);
+}
+
```

Comment:
I probably would just write one function and comment what each test was doing...

Reply:
My biggest problem with our current test organization is that test functions are too long. These one off short tests are easy to isolate, reason about, and run with `-1`.